arangod/Cluster/Maintenance.cpp

```diff
@@ -554,65 +636,110 @@ arangodb::Result arangodb::maintenance::diffPlanLocal(
+        }
       }
     }
   }
 
   // See if shard errors can be thrown out:
+  // Check all shard errors in feature, if database or collection gone -> reset error
+
   for (auto& shard : errors.shards) {
     std::vector<std::string> path = split(shard.first, '/');
-    path.pop_back();           // Get rid of shard
-    if (!pdbs.hasKey(path)) {  // we can drop the local error
-      shard.second.reset();
+    auto const& dbname = path[0];
+    auto const& colname = path[1];
+
+    if (dirty.find(dbname) != dirty.end()) { // only if among dirty
+      auto const planit = plan.find(dbname);
+
+      if (planit == plan.end() ||                        // database gone
+          !planit->second->slice()[0].hasKey(std::vector<std::string>{ // collection gone
+              AgencyCommHelper::path(), PLAN, COLLECTIONS, dbname, colname})) {
+        shard.second.reset();
+      }
     }
   }
 
   // See if index errors can be thrown out:
+  // Check all shard errors in feature, if database, collection or index gone -> reset error
+
   for (auto& shard : errors.indexes) {
-    std::vector<std::string> path = split(shard.first, '/');  // dbname, collection, shardid
-    path.pop_back();             // dbname, collection
-    path.emplace_back(INDEXES);  // dbname, collection, indexes
-    VPackSlice indexes = pdbs.get(path);
-    if (!indexes.isArray()) {  // collection gone, can drop errors
-      for (auto& index : shard.second) {
-        index.second.reset();
-      }
-    } else {  // need to look at individual errors and indexes:
-      for (auto& p : shard.second) {
-        std::string const& id = p.first;
-        bool found = false;
-        for (auto const& ind : VPackArrayIterator(indexes)) {
-          if (ind.get(ID).copyString() == id) {
-            found = true;
-            break;
-          }
+    std::vector<std::string> path = split(shard.first, '/'); // dbname, collection, shardid
+    auto const& dbname = path[0];
+    auto const& colname = path[1];
+
+    if (dirty.find(dbname) != dirty.end()) { // only if among dirty
+
+      auto const planit = plan.find(dbname);
+
+      std::vector<std::string> path{
+        AgencyCommHelper::path(), PLAN, COLLECTIONS, dbname, colname};
+      if (planit == plan.end() ||                        // db gone
+          !planit->second->slice()[0].hasKey(path)) { // collection gone
+        for (auto& index : shard.second) {
+          index.second.reset();
         }
-        if (!found) {
-          p.second.reset();
+      } else {
+        path.push_back(INDEXES);
+        VPackSlice indexes = planit->second->slice()[0].get(path);
+        TRI_ASSERT(indexes.isArray());
+        if (indexes.isArray()) {
+          for (auto& p : shard.second) {
+            std::string const& id = p.first;
+            bool found = false;
+            for (auto const& ind : VPackArrayIterator(indexes)) {
+              if (ind.get(ID).copyString() == id) {
```

Comment:
why `copyString()` instead of `stringView()`?

Reply:
done, thank you